crates/ruff_formatter/src/lib.rs

```diff
@@ -220,13 +202,6 @@ pub trait FormatContext {
 
     /// Returns the formatting options
     fn options(&self) -> &Self::Options;
-
-    /// Returns [None] if the CST has not been pre-processed.
-    ///
-    /// Returns [Some] if the CST has been pre-processed to simplify formatting.
-    /// The source map can be used to map positions of the formatted nodes back to their original
-    /// source locations or to resolve the source text.
-    fn source_map(&self) -> Option<&TransformSourceMap>;
```

Comment:
We may need source maps in the long term to enable range formatting (formatting the selected text) or to suppress formatting of specific nodes